rust/rope/src/tree.rs

```diff
@@ -774,3 +832,43 @@ impl Metric<BytesInfo> for BytesMetric {
 }
 
 */
+
+#[cfg(test)]
+mod test {
+    use ::rope::*;
+    use super::*;
+
+    fn build_triangle(n: u32) -> String {
+        let mut s = String::new();
+        let mut line = String::new();
+        for _ in 0 .. n {
+            s += &line;
+            s += "\n";
+            line += "a";
+        }
+        s
+    }
+
+    #[test]
+    fn cursor_next_triangle() {
+        let n = 10_000;
```

Comment:
This test takes 3s, I think if this were 2_000 it would test it as well and be a lot faster.